packages/webpack-config/webpack/webpack.common.js

```diff
@@ -43,14 +46,28 @@ const mediaLoaderConfiguration = {
   ],
 };
 
-module.exports = function(env) {
+module.exports = function(env = {}) {
   const locations = getLocations(env.projectRoot);
   const babelConfig = createBabelConfig(locations.root);
-  const indexHTML = createIndexHTMLFromAppJSON(locations);
-  const clientEnv = createClientEnvironment(locations);
 
-  const nativeAppManifest = require(locations.appJson);
-  const appManifest = nativeAppManifest.expo || nativeAppManifest;
+  const appJSON = require(locations.appJson);
+  if (!appJSON) {
+    throw new Error('app.json could not be found at: ' + locations.appJson);
+  }
```

Comment:
Expo CLI reads app.json configs using this function: https://github.com/expo/expo-cli/blob/a9ce6ead81f850c9bc5c421fe7a41ac30d9224f2/packages/xdl/src/project/ProjectUtils.js#L212 It handles things like showing errors with code snippets if there's a syntax error in the JSON file (instead of the unhelpful `SyntaxError: app.json: Unexpected token in JSON at position 113` from `require()`). It also sets a couple of default values and supports JSON5 as an alternative.

I think it would make sense to use the same code to load `app.json` configuration everywhere. We can extract the config reading code from `Expo.ProjectUtils` to a small package, e.g. `@expo/config` so the webpack doesn't need to have a (circular) dependency on XDL.

Reply:
https://github.com/expo/expo-cli/pull/452